From c285caba202da593e5faabe16cbe7d8e73254db5 Mon Sep 17 00:00:00 2001
From: Xiaolei Wang <xiaolei.wang@windriver.com>
Date: Fri, 13 Mar 2020 12:45:21 +0800
Subject: [PATCH] ASoC: fsl_sai: Modify the sai->slave_mode[tx] mode

SND_SOC_DAIFMT_CBS_CFS mode should be set to sai-> slave_mode [tx] =
false, otherwise when running audio, clock cannot be configured via
fsl_sai_hw_params,This will cause the system to hang and consume too
much CPU

Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 sound/soc/fsl/fsl_sai.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/sound/soc/fsl/fsl_sai.c b/sound/soc/fsl/fsl_sai.c
index c5752e131d49..ff56be56f870 100644
--- a/sound/soc/fsl/fsl_sai.c
+++ b/sound/soc/fsl/fsl_sai.c
@@ -417,14 +417,12 @@ static int fsl_sai_set_dai_fmt_tr(struct snd_soc_dai *cpu_dai,
 	case SND_SOC_DAIFMT_CBS_CFS:
 		val_cr2 |= FSL_SAI_CR2_BCD_MSTR;
 		val_cr4 |= FSL_SAI_CR4_FSD_MSTR;
-		sai->slave_mode[tx] = true;
 		break;
 	case SND_SOC_DAIFMT_CBM_CFM:
 		sai->slave_mode[tx] = true;
 		break;
 	case SND_SOC_DAIFMT_CBS_CFM:
 		val_cr2 |= FSL_SAI_CR2_BCD_MSTR;
-		sai->slave_mode[tx] = true;
 		break;
 	case SND_SOC_DAIFMT_CBM_CFS:
 		val_cr4 |= FSL_SAI_CR4_FSD_MSTR;
-- 
2.17.1

